lib/agent.js

```diff
@@ -48,22 +101,13 @@ function dispatchFromAgent (requestType) {
       throw new InvalidArgumentError('unsupported opts.path')
     }
 
-    // if url is a string transform into URL inst. otherwise use as is
-    url = typeof url === 'string' ? new URL(url) : url
-    // ensure it atleast has a non-empty string origin
-    if (!url || typeof url.origin !== 'string' || url.origin === '') {
-      throw new InvalidArgumentError('Argument url.origin must be a non-empty string')
-    }
-
     const client = agent.get(url.origin)
 
     if (client && typeof client[requestType] !== 'function') {
       throw new InvalidReturnValueError(`Client returned from Agent.get() does not implement method ${requestType}`)
     }
 
-    const path = url.path || `${url.pathname || '/'}${url.search || ''}`
```

Comment:
You should keep this

Reply:
I'm not super clear on parseOrigin util, but can its result have the `path` prop? If so i'll add that piece back. I've included the right hand side inline of the return statement

Reply:
`${url.pathname || '/'}${url.search || ''}` is sufficient.